src/components/Register.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { Button, Form, Input, Popover } from "antd";
+import { createRegistration } from "@dsnp/sdk";
+
+interface RegisterProps {
+  walletAddress: string;
+  onSuccess: (fromId: string) => void;
+  onFailure: (e: Error) => void;
+}
+
+const Register = (registerOptions: RegisterProps): JSX.Element => {
+  const [
+    createHandleFormVisible,
+    setCreateHandleFormVisible,
+  ] = React.useState<boolean>(true);
+
+  const register = async (formValues: { handle: string }) => {
+    try {
+      const userId = await createRegistration(
+        registerOptions.walletAddress,
+        formValues.handle
+      );
+      if (userId) {
+        registerOptions.onSuccess(userId.replace("dsnp://", ""));
+      }
+      setCreateHandleFormVisible(false);
```

Comment:
could we remove could we remove  setCreateHandleFormVisible(false); since we have it in finally?